plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergSplitSource.java

```diff
@@ -94,7 +116,53 @@ private ConnectorSplit toIcebergSplit(FileScanTask task)
                 task.length(),
                 task.file().fileSizeInBytes(),
                 task.file().format(),
-                ImmutableList.of(),
+                getAddresses(task),
                 getPartitionKeys(task));
     }
+
+    private List<HostAddress> getAddresses(FileScanTask task)
+    {
+        if (fetchSplitLocations) {
+            return blockLocations(task);
+        }
+        else {
+            return ImmutableList.of();
+        }
+    }
```

Comment:
Why not inline `blockLocations` here?

```java
private List<HostAddress> getAddresses(FileScanTask task)
{
    if (!fetchSplitLocations) {
        return List.of();
    }

    // inline blockLocations here
}
```

(I also prefer `List.of()` over `ImmutableList.of()`, but that doesn't really matter.)